### DISCUSSION TITLE
Testing DI when I use Isar.

### DISCUSSION THREAD

**Discussion by alirezat66** [isar/isar#1297] [Help]
Hi everyone, I try to use widget test to test my DI( using Getit)
in my DI I use Isar:
` final dir = await getApplicationDocumentsDirectory();
    late Isar  isar;
    try {
      isar = await Isar.open([WishModelSchema], directory: dir.path);
    } catch (e) {
      isar = Isar.getInstance()!;
    }`
and in bellow you can see my test class:
`
testWidgets('My Test', (WidgetTester tester) async {
    // This line initializes WidgetsBinding
    WidgetsFlutterBinding.ensureInitialized();
    final TestWidgetsFlutterBinding binding =
        TestWidgetsFlutterBinding.ensureInitialized();
    const MethodChannel channel =
        MethodChannel('plugins.flutter.io/path_provider');
    // Change the locale to whatever you want
    binding.setLocale('en', 'US');

    // Now, you can get the locale from WidgetsBinding
    WidgetsBinding.instance.platformDispatcher.locale;
    tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(channel,
        (MethodCall methodCall) async {
      return './test/fixtures/core';
    });

    final locator = GetIt.instance;
    SharedPreferences preferences = MockSharedPreferences();
    when(preferences.getString(prefCurrentMarket)).thenReturn('de');
    when(preferences.containsKey(prefCurrentMarket)).thenReturn(true);
    await ServiceLocator.setup(preferences: preferences);
    expect(locator.isRegistered<AlgoliaQueryBuilder>(), isTrue);

    // ...Rest of your test...
  });
`
when I run the test, I got this error:
Incorrect Isar Core version: Required 3.1.0+1 found 3.0.5. Make sure to use the latest isar_flutter_libs. If you have a Dart only project, make sure that old Isar Core binaries are deleted.
I got the same error when I try to setup isar in my test class too.

**Comment by LesYampolskyi** [comment]
You need to delete a previously downloaded Isar core binaries file <code>libisar.dylib</code> in your root directory and let <code>await Isar.initializeIsarCore(download: true);</code> download a new version.
